### ISSUE TITLE
strings_as_factors_linter false positive for c() with quote-named arguments

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1036]
As pointed out by @AshesITR in #1020, there's a false positive in `strings_as_factors` linter for:

```
data.frame(a = c('b c' = 1))
```
